### PR TITLE
[APB-2774][AK] new page and routing for startOfMapping if HMRC-AS-AGE…

### PR DESCRIPTION
**Pull request by arkowalski**
…NT/ARN

### CODE REVIEW

**Comment by arkowalski** [comment]
squash into 1

File: app/uk/gov/hmrc/agentmappingfrontend/views/start.scala.html [hunk 1]
```diff
@@ -53,21 +53,9 @@ <h2 class="heading-medium">@Messages("connectAgentServices.start.h2")</h2>
 
     <h2 class="heading-medium">@Messages("connectAgentServices.start.beforeYouStart")</h2>
 
-
-    @if(arnOpt.isDefined) {
-        <p class="panel panel-border-wide">@Messages("connectAgentServices.start.l2.1.dynamic-arn", arnOpt.get)</p>
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentmappingfrontend/views/start.scala.html]: If these messages aren't referenced anywhere else in the project, they can be removed from the messages file..

- connectAgentServices.start.l2.1.dynamic-arn
- connectAgentServices.start.l2.1
- connectAgentServices.start.youWillAlsoNeed
- connectAgentServices.start.l2.2
- connectAgentServices.start.l2.3

File: conf/application.conf [hunk 2]
```diff
@@ -86,6 +86,7 @@ microservice {
       company-auth-frontend {
         external-url = "http://localhost:9025"
         sign-in.path = "/gg/sign-in"
+        sign-in.continue-url="/agent-mapping/start"
```
Comment (johnrevill) [inline conf/application.conf]: Does this need the `http://localhost:9438/` preamble to support local dev like the other continue URL below?

File: it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala [hunk 3]
```diff
@@ -31,20 +31,50 @@ class MappingControllerISpec extends BaseControllerISpec with AuthStubs {
       val result = callEndpointWith(request)
       status(result) shouldBe 200
       checkHtmlResultContainsMsgs(result, "connectAgentServices.start.title")
-      bodyOf(result) should include("TARN-000-0001")
     }
 
-    "display the start page for unAuthenticated user" in {
+    "display the /start/sign-in-required for unAuthenticated or user without HMRC-AS-AGENT/ARN" in {
```
Comment (johnrevill) [inline it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala]: This test is just for an unauthenticated user that's not even logged in isn't it?

File: it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala [hunk 4]
```diff
@@ -31,20 +31,50 @@ class MappingControllerISpec extends BaseControllerISpec with AuthStubs {
       val result = callEndpointWith(request)
       status(result) shouldBe 200
       checkHtmlResultContainsMsgs(result, "connectAgentServices.start.title")
-      bodyOf(result) should include("TARN-000-0001")
```
Comment (johnrevill) [inline it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala]: Page no longer shows the ARN, so test name should be updated.

File: conf/application.conf (hunk 2 above)
Comment (arkowalski) [reply conf/application.conf]: yeah, i have picked up on this, it was sitting in the frontendAppConfig instead

**Pull request merged**